tests/claws-container/vect-test.cpp

```diff
@@ -0,0 +1,137 @@
+#include <gtest/gtest.h>
+#include <type_traits>
+#include <claws/container/vect.hpp>
+
+TEST(vect, member_types)
+{
+  using vec = claws::vect<int, 2>;
+
+  static_assert(std::is_same_v<typename vec::value_type, int>);
+  static_assert(std::is_same_v<typename vec::reference, int &>);
+  static_assert(std::is_same_v<typename vec::const_reference, const int &>);
+  static_assert(std::is_same_v<typename vec::pointer, int *>);
+  static_assert(std::is_same_v<typename vec::const_pointer, const int *>);
+  static_assert(std::is_same_v<typename vec::const_pointer, const int *>);
+}
+
+// Array iterators are pointers, so they cannot be constexpr for stack-local variables.
+// However, they are for static variables.
+static constexpr claws::vect<int, 2> static_vect{{1, 2}};
+
+TEST(vect, constexpr_correctness)
+{
+  constexpr claws::vect<int, 2> v1{{1, 2}};
+
+  static_assert(v1[0] == 1);
+  static_assert(v1[1] == 2);
+
+  static_assert(v1.x() == 1);
+  static_assert(v1.y() == 2);
+
+  static_assert(v1.size() == 2);
+  static_assert(v1.max_size() == 2);
+  static_assert(!v1.empty());
+
+  constexpr auto it = static_vect.begin();
+  static_assert(*it == 1);
+
+  static_assert(static_vect == static_vect);
+  static_assert(!(static_vect < static_vect));
+  static_assert(!(static_vect > static_vect));
+  static_assert(static_vect <= static_vect);
+  static_assert(static_vect >= static_vect);
+}
+
+template<typename T, typename U>
+static constexpr const bool check_same_noexceptness = std::is_nothrow_destructible_v<
+                                                        T> == std::is_nothrow_destructible_v<U> &&std::is_nothrow_move_constructible_v<T> == std::is_nothrow_move_constructible_v<U> &&std::is_nothrow_copy_constructible_v<T> == std::is_nothrow_copy_constructible_v<U> &&std::is_nothrow_move_assignable_v<T> == std::is_nothrow_move_assignable_v<U> &&std::is_nothrow_copy_assignable_v<T> == std::is_nothrow_copy_assignable_v<U>;
+
+namespace
+{
+  class throw_copy_constructible
+  {
+  public:
+    throw_copy_constructible(const throw_copy_constructible &) noexcept(false)
+    {}
+  };
+}
+
+TEST(vect, noexcept_correctness)
+{
+  static_assert(check_same_noexceptness<int, claws::vect<int, 3>>);
+  static_assert(check_same_noexceptness<throw_copy_constructible, claws::vect<throw_copy_constructible, 3>>);
+}
+
+TEST(vect, construction)
+{
+  constexpr int array[] = {5, 4, 3, 2, 1};
+  constexpr claws::vect v1(array);
+  constexpr claws::vect v2(v1);
+  constexpr claws::vect<long, 5> v3(v2);
+  constexpr claws::vect<int, 3> v4;
+  static_assert(v4[0] == 0);
+  static_assert(v4[1] == 0);
+  static_assert(v4[2] == 0);
+  constexpr claws::vect<int, 4> vec5(1, 2, 3, 4);
+  constexpr claws::vect<int, 4> vec6{1, 2, 3, 4};
+}
+
+TEST(vect, copy)
+{
+  constexpr claws::vect<int, 5> from{{1, 2, 3, 4, 5}};
+  claws::vect<int, 5> to(from);
```

Comment:
It used to be possible to have the `to` `vect` be `constexpr` too. This seems like a regression to me.

Reply:
It actually can be constexpr here (see the `construction` test case). This test case is actually not really needed anymore then.

Reply:
Ah, nice. Please remove it then.